### PR TITLE
Simplify /info endpoint for custom connectors

### PR DESCRIPTION
**Pull request by zregvart**
**Based on #495 and #500 please review that first**

This removes `/api/v1/custom/connectors/{id}/validation` endpoint and changes the response from `/api/v1/custom/connectors/{id}/info` to be a bit simpler and in the same time to contain errors and warnings.

//cc @deeleman

### CODE REVIEW

**Comment by rhuss** [comment]
@zregvart could you please rebase ? I will review it the right after, should be the last PR to review for the connectors so far, right ?

**Comment by zregvart** [comment]
@rhuss should be ok to review, thanks 💯

**Comment by zregvart** [comment]
> should be the last PR to review for the connectors so far, right ?

yes, indeed

File: rest/model/src/main/java/io/syndesis/model/Violation.java [hunk 1]
```diff
@@ -13,7 +13,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package io.syndesis.rest.v1.operations;
+package io.syndesis.model;
```
Comment (rhuss) [inline rest/model/src/main/java/io/syndesis/model/Violation.java]: what's the difference between "message" and "error" in Violation ?

Comment (rhuss) [reply rest/model/src/main/java/io/syndesis/model/Violation.java]: in fromConstraintViolation() is this all null safe ? (sorry can not comment directly on the code).

**Comment by rhuss** [comment]
@zregvart there are compile errors ....

**Comment by zregvart** [comment]
@rhuss should be OK now, and this is why you don't have long running branches

File: rest/model/src/main/java/io/syndesis/model/Violation.java (hunk 1 above)
Comment (zregvart) [reply rest/model/src/main/java/io/syndesis/model/Violation.java]: The `message` holds the text of an error (`"XYZ is required"`), and error contains the class of error (`"NotEmpty"`).

Should be null safe, it's the way bean validation works, constraints are applied via annotations, and each violation has a resolved message (or a placeholder message), if that's what you mean.

**Pull request merged**